aggregator/src/cache.rs

```diff
@@ -206,3 +210,264 @@ impl PeerAggregatorCache {
             .find(|peer| peer.endpoint() == endpoint && peer.role() == role)
     }
 }
+
+#[derive(Debug)]
+pub struct TaskAggregatorCache<C: Clock> {
+    clock: C,
+    datastore: Arc<Datastore<C>>,
+    report_writer: Arc<ReportWriteBatcher<C>>,
+    entries: StdMutex<HashMap<TaskId, Arc<Mutex<TaskAggregatorCacheEntry<C>>>>>,
+    taskprov_enabled: bool,
+    ttl_seconds: Uniform<u64>,
+}
+
+#[derive(Debug)]
+struct TaskAggregatorCacheEntry<C: Clock> {
+    expiration: Time,
+    is_initialized: bool,
```

Comment:
golfing: I think we _could_ get rid of `is_initialized` entirely, based on the argument in https://github.com/divviup/janus/pull/3029/files#r1571390249.

Otherwise, instead of a flag, we could express this state in a different way: I think the best way would be for `TaskAggregatorCache::entries`' value type to be `Arc<Mutex<Option<TaskAggregatorCacheEntry>>>`, with `None` inside the mutex indicating a lack of initialization.

(Other options: either `expiration` could be an `Option<Time>` (or maybe just set it to a time far in the past, though this is hacky), or maybe `task_aggregator` could be an `Option<Option<TaskAggregator>>`.)

The reason I think these approaches are nicer is that, from a design perspective, the boolean flag allows "invalid" states to be expressed (e.g. `is_initialized = false`, but a value is in place), while the recommended approaches do not (to varying degrees). N.B. I don't think these invalid states can be encountered with the code as it stands, this is more future-proofing against bugs.